core/appeal/service_test.go

```diff
@@ -697,18 +717,51 @@ func (s *ServiceTestSuite) TestCreate() {
 				},
 				AppealConfig: &domain.PolicyAppealConfig{AllowOnBehalf: true},
 			},
+			{
```

Comment:
Can you add a test case, where a field in IAM response was used in appeal steps - but the field itself was not present in IAM response.

Example:

sample policy step - 
```
    approvers:
      - "$appeal.creator.manager_email != nil && len($appeal.creator.manager_email) > 3 ? $appeal.creator.manager_email : 'test-abc@email.com'"

```

and IAM response did not have `manager_email` field altogether for a given request.

Reply:
this returns `cannot fetch manager_email from <nil>` error from the expr library. But this should be considered expected as we are not doing `$appeal.creator != nil` check in the expression string 🤔

Reply:
Not really, as even if we do this - `$appeal.creator != nil`, the creator details are present, its just that the json response might not always have all the expected fields.

example:
IAM config schema:
    department: department
    team_name: teamName
    job_title: title
    manager_email: managerEmail
    org_name: OrgName
    requestor_name: fullName

In the response if all other fields are present, and only `manager_email` field is not present in json - then still there would be error.

Reply:
@bsushmith in that case, it's more related to nil evaluation value handling, previously it was returning a reflection error but already got fixed in another PR #32. 

changes in this PR only for allowing error from iam fetch then setting `creator` value to `nil`. Expression evaluation shouldn't get affected here